src/webview_candidate_window.mm

```diff
@@ -36,6 +36,74 @@ - (BOOL)isKeyWindow {
 
 @end
 
+#ifdef WKWEBVIEW_PROTOCOL
+@implementation FileSchemeHandler
+
+// In f5m, it serves fcitx:///file/foo/bar from
+// ~/.local/share/fcitx5/www/foo/bar
+- (void)webView:(WKWebView *)webView
+    startURLSchemeTask:(id<WKURLSchemeTask>)urlSchemeTask {
+    NSURL *url = urlSchemeTask.request.URL;
+    if (!url || ![url.path hasPrefix:@"/file"]) {
+        [urlSchemeTask didFailWithError:[NSError errorWithDomain:@"Invalid URL"
+                                                            code:0
+                                                        userInfo:nil]];
+        return;
+    }
+    NSString *filePath = [url.path substringFromIndex:[@"/file" length]];
+    std::cerr << "Accessing " << [filePath UTF8String] << " from webview"
```

Comment:
leftover debug print?

Reply:
By design. Later users will have more control in the content of HTML. As I cannot guarantee this is absolutely secure, I'd at least log what is accessed for now.

Reply:
👍 good point